src/AasxServerStandardBib/Services/AdminShellPackageEnvironmentService.cs

```diff
@@ -599,49 +600,97 @@ public List<ISubmodel> GetAllSubmodels()
-            //Check if Submodel exists
-            var found = IsSubmodelPresent(newSubmodel.Id, out _, out _);
-            if (found)
+            bool found = true;
+
+            //check if submodel is already referenced in any of the shells
+            var foundPackages = new Dictionary<int, List<IAssetAdministrationShell>>();
+            for(int i = 0;  i< _packages.Length; i++)
             {
-                throw new DuplicateException($"Submodel with id {newSubmodel.Id} already exists.");
+                var package = _packages[i];
+                if (package != null)
+                {
+                    var env = package.AasEnv;
+                    // Check if the Submodel is already present in the package
+                    // If yes, then skip the iteration, it will be handled by Blazor
+                    var isSmAlreadyPresent = env.Submodels!.Exists(s => s.Id.Equals(newSubmodel.Id));
+                    if (isSmAlreadyPresent)
+                    {
+                        found = true;
+                    }
+                    if (env != null && !env.AssetAdministrationShells.IsNullOrEmpty() && !isSmAlreadyPresent)
+                    {
+                        var foundAAS = new List<IAssetAdministrationShell>();
+                        foreach (var aas in env.AssetAdministrationShells!)
+                        {
+                            if (aas != null && !aas.Submodels.IsNullOrEmpty())
+                            {
+                                var smFound = aas.Submodels.Exists(s => s.GetAsIdentifier().Equals(newSubmodel.Id));
+                                if(smFound)
+                                {
+                                    foundAAS.Add(aas);
+                                }
+                            }
+                        }
+                        if(foundAAS.Any())
+                        {
+                            foundPackages.Add(i, foundAAS);
+                        }
+                    }
+                }
             }
 
-            //Check if corresponding AAS exist. If yes, then add to the same environment
-            if (!string.IsNullOrEmpty(aasIdentifier))
+            if(foundPackages.Count > 0)
             {
-                var aasFound = IsAssetAdministrationShellPresent(aasIdentifier, out IAssetAdministrationShell aas, out int packageIndex);
-                if (aasFound)
+                foreach (var package in foundPackages)
                 {
-                    newSubmodel.SetAllParents(DateTime.UtcNow);
-                    aas.Submodels ??= new List<IReference>();
-                    aas.Submodels.Add(newSubmodel.GetReference());
-                    _packages[packageIndex].AasEnv.Submodels.Add(newSubmodel);
+                    var packageIndex = package.Key;
+                    _packages[packageIndex].AasEnv!.Submodels ??= new List<ISubmodel>();
+                    _packages[packageIndex].AasEnv!.Submodels!.Add(newSubmodel);

```

Comment:
Unless I'm missing something obvious, this code path doesn't make a lot sense.

The old behavior was to add the submodel directly to the aasIdentifier, without needing to append the submodel to the AAS submodel list. Of course this isn't possible without an aasIdentifier anymore.

However, now `foundPackages` is the list of AASs that already include the new submodel identifier and yet we add the submodel to the AAS submodel list. This results in the submodel entry being duplicated. Usually this doesn't happen because the submodel is new and not found in any AAS anyway.